source/AI.cpp

```diff
@@ -1736,19 +1811,16 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
 			parent.Attributes().Get("fuel capacity") - maxFuel)
 		return true;
 
-	// If an out-of-combat NPC carried ship is carrying a significant cargo
+	// If an out-of-combat carried ship is carrying a significant cargo
 	// load and can transfer some of it to the parent, it should do so.
-	if(!ship.IsYours())
```

Comment:
Not having this is a pain for a player who is mining rather than plundering or capping.  They have to _carefully maneuver_  escort fighters in a certain position and then they have to carefully maneuver their flagship over top.  They have to painstakingly figure out which escort they are hovered over at the moment in order to dump the right cargo.  One player even tried dumping cargo while a fighter was recalled and [hit a game bug](https://github.com/endless-sky/endless-sky/pull/6723) because nobody does that normally.

Not having this as a feature makes fighters impractical for mining.  However, due to speed fighters are ideal ships to be mining.  Is there a compromise we can come to for this such as a setting?  If not a setting can we check if the flagship has `asteroid scanning` to enable the behavior?

Reply:
I pushed a change for conditional cargo transfer.  Cargo will only be auto transferred if the player flagship has asteroid scanning.  Can you let me know if this is acceptable?  Another alternative could be a setting (though I don't like lots of settings so I hesitate to add one).

Reply:
2027b9bc10ae6dfc4cf869818d400eaccb314d72 now a preference